main/pipeline/pipeline.cc

```diff
@@ -999,7 +1005,6 @@ ast::ParsedFilesOrCancelled typecheck(unique_ptr<core::GlobalState> &gs, vector<
                      !result.done();
                      result = resultq->wait_pop_timed(threadResult, WorkerPool::BLOCK_INTERVAL(), gs->tracer())) {
                     if (result.gotItem()) {
-                        counterConsume(move(threadResult.counters));
```

Comment:
I think you can remove the `typecheck_thread_result` type btw; we no longer need it. It was created to store the ASTs + counters.